### ISSUE TITLE
What are these training configurations of downstream tasks in SUPERB paper?

### ISSUE THREAD

**Issue by mechanicalsea** [s3prl/s3prl#268]
The training configurations of downstream tasks were not reported in the original SUPERB paper. If they are provided, it can help followers step into the SUPERB benchmark.

I got into trouble while reproducing HuBERT SID by a batch size of 32 and a learning rate of 1e-4. I found that using the different batch sizes and learning rates can result in significantly different performances. For a fair comparison, it can be helpful if the training configurations used in the SUPERB paper are released. WavLM also reported different batch sizes and learning rates.

Thank you.

**Comment by leo19941227** [comment]
Hi!

Thanks for the great suggestion!
This will really help people more easily reproduce our results.
I would document about how to get our finetuning config for reproducing.

In SUPERB, we use the exactly same config in each downstream folder eg. `downstream/voxceleb1/config.yaml`, including batch_size, except the learning rate. So, we did not tune the batch size for each upstream and keep it the same.
We tune the learning rate across log-scale, and here are some results for HuBERT Base:

| learning rate | ACC |
| - | - |
| 1e-1 | 0.814204335 |
| 1e-2 | 0.810083627 |
| 1e-3 | 0.7008847 |
| 1e-4 | 0.673736512 |

Hence, I think for lr 1e-4 we really can get good results on hubert-SID.
As for the other upstream/tasks, since I think people might be more interested in reproducing wav2vec2/hubert results, so I have released the [finetuned artifacts](https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/docs/superb_artifacts.md) as the reference.
Despite each downstream/upstream/ folder in the [artifacts server](http://140.112.21.28:8000/) includes only the checkpoints and the Tensorboard events, you can extract the finetuning config from the checkpoint and see how we finetune for a specific upstream-downstream pair.

Specifically,

```bash
#!/bin/bash

s3prl_path="/home/leo/d/Benchmarking/s3prl/s3prl"
working_dir=$(pwd)

wget -r http://140.112.21.28:8000/sid/hubert_1e-1/
cd $s3prl_path
python3 utility/extract_settings.py --ckpt_pth $working_dir/"140.112.21.28:8000/sid/hubert_1e-1/dev-best.ckpt" --save_dir $working_dir
```

Then, you can see the `$working_dir/args.yaml` and `$working_dir/config.yaml` files which contains all the finetuning configs.
For the learning rate, you can see the `optimizer` field in the `config.yaml` file.

I will put the above procedure in the SUPERB documentation. Thanks for the suggestion!

**Comment by leo19941227** [comment]
Or, you can directly infer the best learning rate as shown by the folder name in the artifact server lol
I believe the folder name is correct about the learning rate, but if it is not consistent with the configs saved in the checkpoints, you should treat the checkpoint configs as the truth. (Please help report the inconsistency if you fine one. Thanks!)

**Comment by mechanicalsea** [comment]
The way that extracts configuration from checkpoints is good for me. Thank you for your help. It, indeed, helps us obtain good results.

Have a nice day.

**Comment by amitmeghanani** [comment]
Hello,
I am trying to access the the superb artefacts server but the link is not working. Also, trying to reproduce the results from from the default downstream task configs but getting slightly better or worse results. Could you please help with that?
